### PR TITLE
Primary constructor implement

### PR DESCRIPTION
**Pull request by KhanbalaRashidov**

### CODE REVIEW

File: src/Clean.Architecture.Infrastructure/Data/AppDbContext.cs [hunk 1]
```diff
@@ -4,16 +4,10 @@
 using Microsoft.EntityFrameworkCore;
 
 namespace Clean.Architecture.Infrastructure.Data;
-public class AppDbContext : DbContext
+public class AppDbContext(DbContextOptions<AppDbContext> _options,
+  IDomainEventDispatcher? _dispatcher) : DbContext(_options)
```
Comment (ardalis) [inline src/Clean.Architecture.Infrastructure/Data/AppDbContext.cs:8]: These should not have `_` prefix.

File: src/Clean.Architecture.Infrastructure/Data/EfRepository.cs [hunk 2]
```diff
@@ -4,9 +4,7 @@
 namespace Clean.Architecture.Infrastructure.Data;
 
 // inherit from Ardalis.Specification type
-public class EfRepository<T> : RepositoryBase<T>, IReadRepository<T>, IRepository<T> where T : class, IAggregateRoot
+public class EfRepository<T>(AppDbContext _dbContext) :
```
Comment (ardalis) [inline src/Clean.Architecture.Infrastructure/Data/EfRepository.cs:7]: Remove `_` prefix on constructor parameters.

File: src/Clean.Architecture.Infrastructure/Email/MimeKitEmailSender.cs [hunk 3]
```diff
@@ -6,18 +6,10 @@
 
 namespace Clean.Architecture.Infrastructure.Email;
 
-public class MimeKitEmailSender : IEmailSender
+public class MimeKitEmailSender(ILogger<MimeKitEmailSender> _logger,
+  IOptions<MailserverConfiguration> _mailserverOptions) : IEmailSender
```
Comment (ardalis) [inline src/Clean.Architecture.Infrastructure/Email/MimeKitEmailSender.cs:10]: Remove `_` prefix

File: src/Clean.Architecture.Infrastructure/Email/SmtpEmailSender.cs [hunk 4]
```diff
@@ -9,17 +9,10 @@ namespace Clean.Architecture.Infrastructure.Email;
 /// MimeKit is recommended over this now:
 /// https://weblogs.asp.net/sreejukg/system-net-mail-smtpclient-is-not-recommended-anymore-what-is-the-alternative
 /// </summary>
-public class SmtpEmailSender : IEmailSender
+public class SmtpEmailSender(ILogger<SmtpEmailSender> _logger,
```
Comment (ardalis) [inline src/Clean.Architecture.Infrastructure/Email/SmtpEmailSender.cs:12]: Remove `_`

**Review by ardalis — CHANGES_REQUESTED** [review]

File: src/Clean.Architecture.Infrastructure/Data/AppDbContext.cs (hunk 1 above)
Comment (KhanbalaRashidov) [reply src/Clean.Architecture.Infrastructure/Data/AppDbContext.cs:8]: Okay

File: src/Clean.Architecture.Infrastructure/Email/MimeKitEmailSender.cs (hunk 3 above)
Comment (KhanbalaRashidov) [reply src/Clean.Architecture.Infrastructure/Email/MimeKitEmailSender.cs]: Resolve

File: src/Clean.Architecture.Infrastructure/Email/MimeKitEmailSender.cs [hunk 5]
```diff
@@ -6,22 +6,14 @@
...
-    _logger = logger;
-    _mailserverConfiguration = mailserverOptions.Value!;
-  }
-
+  private readonly MailserverConfiguration _mailserverConfiguration = mailserverOptions.Value!;
 
   public async Task SendEmailAsync(string to, string from, string subject, string body)
   {
-    _logger.LogWarning("Sending email to {to} from {from} with subject {subject} using {type}.", to, from, subject, this.ToString());
+    logger.LogWarning("Sending email to {to} from {from} with subject {subject} using {type}.", to, from, subject, this.ToString());
```
Comment (ardalis) [inline src/Clean.Architecture.Infrastructure/Email/MimeKitEmailSender.cs:16]: Avoid using class level parameters directly - assign to _logger and use that.

File: src/Clean.Architecture.Infrastructure/Email/SmtpEmailSender.cs [hunk 6]
```diff
@@ -33,6 +26,6 @@ public async Task SendEmailAsync(string to, string from, string subject, string
     };
     message.To.Add(new MailAddress(to));
     await emailClient.SendMailAsync(message);
-    _logger.LogWarning("Sending email to {to} from {from} with subject {subject} using {type}.", to, from, subject, this.ToString());
+    logger.LogWarning("Sending email to {to} from {from} with subject {subject} using {type}.", to, from, subject, this.ToString());
```
Comment (ardalis) [inline src/Clean.Architecture.Infrastructure/Email/SmtpEmailSender.cs:29]: See above.

**Review by ardalis — CHANGES_REQUESTED** [review]

File: src/Clean.Architecture.Infrastructure/Email/MimeKitEmailSender.cs (hunk 5 above)
Comment (ardalis) [inline src/Clean.Architecture.Infrastructure/Email/MimeKitEmailSender.cs:16]: this.ToString() should really be `body`

**Comment by ardalis** [comment]
Thanks!

**Pull request merged**